tests/lib/tools/suite/journal-state/task.yaml

```diff
@@ -13,20 +13,30 @@ execute: |
     # Check that the test is correctly started in the journal log
     "$TESTSTOOLS"/journal-state start-new-log
     echo "Add some extra logs" | systemd-cat -t snapd-test
-    retry --wait 1 --attempts 30 sh -c "test $(journalctl --cursor "$cursor1" | grep -c "$SPREAD_JOB") -eq 1"
+    # Here cannot be used the retry tool because we need to evaluate the whole
+    # test command within the retry but the cursor1 var within the test
```

Comment:
Ahh, I got it. Your fix is correct but I wonder if proper quoting of arguments to retry wouldn't fix it? Although from quick look I'm not sure subprocess.call in python will do proper expanstion, cc @zyga

Reply:
How about:
```
retry -n 10 sh -c 'test $(journalctl --cursor '"$cursor1"' | grep -c '"$SPREAD_JOB"') -eq 1'
```

Reply:
@zyga Thanks for the suggestion, finally what worked was: 
retry --wait 1 --attempts 10 sh -c 'test $(journalctl --cursor '\'"$cursor1"\'' | grep -c '"$SPREAD_JOB"') -eq 1'